src/HTTP Server/main.vala

```diff
@@ -0,0 +1,85 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1 --pkg libsoup-3.0
+private Gtk.Label label_greetings;
+private Gtk.ToggleButton button_server;
+private Gtk.LinkButton linkbutton;
+int port;
+
+void handler (Soup.Server server, Soup.ServerMessage msg, string path, GLib.HashTable<string, string>? query) {
+    var content_type_params = new GLib.HashTable<string, string> (str_hash, str_equal);
+    content_type_params.insert ("charset", "UTF-8");
```

Comment:
Vala has some nice syntactic sugar for methods named `set` and `get`, so you can use them just like array indices
```suggestion
    content_type_params["charset"] = "UTF-8";
```

For more info: https://wiki.gnome.org/Projects/Vala/Tutorial#Methods_With_Syntax_Support

Reply:
Those are nice :). Fixed, thank you